R/feature.align.R

```diff
@@ -1,178 +1,194 @@
-            text(x=0,y=0,mz.tol,cex=1.2)
+            plot(c(-1, 1), c(-1, 1), type = "n", xlab = "", ylab = "",
+                 main = "alignment m/z tolerance level given", axes = FALSE)
+            text(x = 0, y = 0, mz_tol_relative, cex = 1.2)
         }
         
-        if(!is.na(chr.tol) && do.plot)
-        {
-            plot(c(-1,1),c(-1,1),type="n",xlab="",ylab="",main="retention time \n tolerance level given",axes=FALSE)
-            text(x=0,y=0,chr.tol,cex=1.2)
+        if (!is.na(rt_tol_relative) && do.plot) {
+            plot(c(-1, 1), c(-1, 1), type = "n", xlab = "", ylab = "",
+                 main = "retention time \n tolerance level given", axes = FALSE)
+            text(x = 0, y = 0, rt_tol_relative, cex = 1.2)
         }
         
-        all.ft<-find.tol.time(masses, chr, lab, num.exp=num.exp, mz.tol=mz.tol, chr.tol=chr.tol, max.mz.diff=max.align.mz.diff, do.plot=do.plot)
-        chr.tol<-all.ft$chr.tol
+        # find relative retention time tolerance level
+        all.ft <- find.tol.time(mz_values,
+                                chr,
+                                lab,
+                                number_of_experiments = number_of_experiments,
+                                mz_tol_relative = mz_tol_relative,
+                                rt_tol_relative = rt_tol_relative,
+                                mz_tol_absolute = mz_tol_absolute,
+                                do.plot = do.plot)
+        rt_tol_relative <- all.ft$chr.tol
         
         message("**** performing feature alignment ****")
-        message(paste("m/z tolerance level: ", mz.tol))
-        message(paste("time tolerance level:", chr.tol))
+        message(paste("m/z tolerance level: ", mz_tol_relative))
+        message(paste("time tolerance level:", rt_tol_relative))
         
-        aligned.ftrs<-pk.times<-rep(0,4+num.exp)
-        mz.sd.rec<-0
+        aligned.ftrs <- pk.times <- rep(0, 4 + number_of_experiments)
+        mz.sd.rec <- 0
         
-        labels<-unique(all.ft$grps)
+        labels <- unique(all.ft$grps)
         
-        area<-grps<-masses
+        area <- grps <- mz_values
         
+        # grouping the features based on their m/z values (assuming the tolerance level)
         sizes <- c(0, cumsum(sapply(features, nrow)))
-        for(i in 1:num.exp)
-        {
-            this<-features[[i]]
-            sel<-which(all.ft$lab == i)
-            that<-cbind(all.ft$mz[sel], all.ft$chr[sel], all.ft$grps[sel])
-            this<-this[order(this[,1], this[,2]), ]
-            that<-that[order(that[,1], that[,2]), ]
+        for (i in 1:number_of_experiments) {
+            this <- features[[i]]
+            sel <- which(all.ft$lab == i)
+            that <- cbind(all.ft$mz[sel], all.ft$chr[sel], all.ft$grps[sel])
+            this <- this[order(this[, 1], this[, 2]),]
+            that <- that[order(that[, 1], that[, 2]),]
```

Comment:
@zargham-ahmad wasn't exactly the same code also in adjust.time.R?

Reply:
not the whole for loop code, but these 6 lines are exactly the same.